Core/src/org/sleuthkit/autopsy/healthmonitor/ServicesHealthMonitor.java

```diff
@@ -0,0 +1,767 @@
+                System.out.println("  query: " + createCommand);
+                rs = statement.executeQuery(createCommand);
+                if(rs.next()) {
+                    logger.log(Level.INFO, "Existing Services Health Monitor database found");
+                    return true;
+                }
+            } finally {
+                if(rs != null) {
+                    rs.close();
+                }
+            }
+        } catch (UserPreferencesException | ClassNotFoundException | SQLException ex) {
+            throw new HealthMonitorException("Failed check for health monitor database", ex);
+        }
+        return false;
+    }
+    
+    /**
+     * Create a new health monitor database.
+     * @throws HealthMonitorException 
+     */
+    private void createDatabase() throws HealthMonitorException {
+        try {
+            // Use the same database settings as the case
+            CaseDbConnectionInfo db = UserPreferences.getDatabaseConnectionInfo();
+            Class.forName("org.postgresql.Driver"); //NON-NLS
+            try (Connection connection = DriverManager.getConnection("jdbc:postgresql://" + db.getHost() + ":" + db.getPort() + "/postgres", db.getUserName(), db.getPassword()); //NON-NLS
+                    Statement statement = connection.createStatement();) {
+                String createCommand = "CREATE DATABASE \"" + DATABASE_NAME + "\" OWNER \"" + db.getUserName() + "\""; //NON-NLS
+                statement.execute(createCommand);
+            }
+            logger.log(Level.INFO, "Created new health monitor database " + DATABASE_NAME);
+        } catch (UserPreferencesException | ClassNotFoundException | SQLException ex) {
+            throw new HealthMonitorException("Failed to delete health monitor database", ex);
+        }
+    }
+
+    /**
+     * Setup a connection pool for db connections.
+     * @throws HealthMonitorException 
+     */
+    private void setupConnectionPool() throws HealthMonitorException {
+        try {
+            CaseDbConnectionInfo db = UserPreferences.getDatabaseConnectionInfo();
+        
+            connectionPool = new BasicDataSource();
+            connectionPool.setDriverClassName("org.postgresql.Driver");
+
+            StringBuilder connectionURL = new StringBuilder();
+            connectionURL.append("jdbc:postgresql://");
+            connectionURL.append(db.getHost());
+            connectionURL.append(":");
+            connectionURL.append(db.getPort());
+            connectionURL.append("/");
+            connectionURL.append(DATABASE_NAME);
+
+            connectionPool.setUrl(connectionURL.toString());
+            connectionPool.setUsername(db.getUserName());
+            connectionPool.setPassword(db.getPassword());
+
+            // tweak pool configuration
+            connectionPool.setInitialSize(3); // start with 3 connections
+            connectionPool.setMaxIdle(CONN_POOL_SIZE); // max of 10 idle connections
```

Comment:
There should only ever be 1 connection right? If so, might want to consider changing the initial size and max idle numbers.

Reply:
No there can be multiple connections. The timer can start a database write as can the listener for case closing. Additionally displaying the metrics will use a database connection.

Reply:
Ok.